src/test/java/guitests/guihandles/TaskCardViewHandle.java

```diff
@@ -0,0 +1,326 @@
+
+    /**
+     * Constructs a handle for {@link seedu.todo.ui.view.TaskCardView}.
+     *
+     * @param guiRobot The GUI test robot.
+     * @param primaryStage The main stage that is executed from the application's UI.
+     * @param rootNode Node that houses the contents of this Task Card.
+     */
+    public TaskCardViewHandle(GuiRobot guiRobot, Stage primaryStage, Node rootNode){
+        super(guiRobot, primaryStage, null);
+        this.rootNode = rootNode;
+    }
+
+    /* Task Property Getters */
+    public String getDisplayedTitle() {
+        return getTextFromLabel(TITLE_LABEL_ID);
+    }
+
+    public String getDisplayedDescription() {
+        return getTextFromLabel(DESCRIPTION_LABEL_ID);
+    }
+
+    public String getDisplayedDateText() {
+        return getTextFromLabel(DATE_LABEL_ID);
+    }
+
+    public String getDisplayedLocation() {
+        return getTextFromLabel(LOCATION_LABEL_ID);
+    }
+
+    public String getDisplayedTypeLabel() {
+        return getTextFromLabel(TYPE_LABEL_ID);
+    }
+
+    public boolean getMoreInfoLabelVisibility() {
+        Node moreInfoLabel = getNode(MOREINFO_LABEL_ID);
+        return UiTestUtil.isDisplayed(moreInfoLabel);
+    }
+
+    public boolean getDescriptionBoxVisibility() {
+        Node descriptionBox = getNode(DESCRIPTION_BOX_ID);
+        return UiTestUtil.isDisplayed(descriptionBox);
+    }
+
+    public boolean getDateBoxVisibility() {
+        Node dateBox = getNode(DATE_BOX_ID);
+        return UiTestUtil.isDisplayed(dateBox);
+    }
+
+    public boolean getLocationBoxVisibility() {
+        Node locationBox = getNode(LOCATION_BOX_ID);
+        return UiTestUtil.isDisplayed(locationBox);
+    }
+
+    public boolean getPinImageVisibility() {
+        Node pinImage = getNode(PIN_IMAGE_ID);
+        return UiTestUtil.isDisplayed(pinImage);
+    }
+
+    public boolean isSelectedStyleApplied() {
+        return UiTestUtil.containsStyleClass(rootNode, "selected");
+    }
+
+    public boolean isCompletedStyleApplied() {
+        return UiTestUtil.containsStyleClass(rootNode, "completed");
+    }
+
+    public boolean isOverdueStyleApplied() {
+        return UiTestUtil.containsStyleClass(rootNode, "overdue");
+    }
+
+    /* General Methods */
+    /**
+     * Checks if the supplied index matches to what this node displays.
+     * To be used for nodes filtering.
+     */
+    public boolean matchesTask(int displayedIndex) {
+        return getDisplayedTitle().startsWith(displayedIndex + ". ");
+    }
+
+    /**
+     * Given an {@link ImmutableTask} and a displayed index, check if this view is displayed correctly.
+     * @param displayedIndex Index displayed in the view.
+     * @param task Task displayed in the view.
+     * @return Returns true only if and only if the elements in this view is displayed correctly.
+     * TODO: Maybe we do not need this at all.
+     */
+    public boolean isDisplayedCorrectly(int displayedIndex, ImmutableTask task) {
```

Comment:
It feels like this should actually be moved to a lower level, like a unit test for TaskCard, for example. It's okay for this to be here for now